lib/routes/icable/news.ts

```diff
@@ -0,0 +1,243 @@
+import { getCurrentPath } from '@/utils/helpers';
+import path from 'node:path';
+import { art } from '@/utils/render';
+import timezone from '@/utils/timezone';
+import { config } from '@/config';
+
+const __dirname = getCurrentPath(import.meta.url);
+
+export const route: Route = {
+    path: '/news/:category?',
+    categories: ['traditional-media'],
+    example: '/icable/news',
+    parameters: { category: '分類，默認為新聞資訊' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['www.i-cable.com'],
+            target: '/news',
+        },
+        {
+            source: ['www.i-cable.com/category/:category'],
+            target: '/news/:category',
+        },
+    ],
+    name: '新聞',
+    maintainers: ['quiniapiezoelectricity'],
+    handler,
+    url: 'www.i-cable.com/',
+    description: `:::tip
+    分類可用分類名稱或ID，如：新聞資訊或是38
+    如需RSS內嵌影音，請在URL後加上?embed=1，否則默認以圖片代替
+  :::`,
+};
+
+function getMainCategoryQuery() {
+    return {
+        query: `
+  query {
+    mainCategory {
+      id
+      name
+      description
+    }
+  }
+`,
+    };
+}
+
+function getCategoryEdgesQuery(mainCategoryId) {
+    return {
+        query: `
+    query GetCategoryEdges($_id: Int) {
+        categories(where: { parent: $_id, hideEmpty: true }) {
+          edges {
+            node {
+              id
+              databaseId
+              name
+            }
+          }
+        }
+      }
+    `,
+        variables: {
+            _id: mainCategoryId,
+        },
+        operationName: 'GetCategoryEdges',
+    };
+}
+
+function getListsByCateQuery(categoryId, limit) {
+    return {
+        query: `
+  query getListsByCate($_category: Int) {
+    posts(first: ${limit}, after: "", where:{categoryId: $_category}) {
+      pageInfo {
+        hasPreviousPage
+        startCursor
+        hasNextPage
+        endCursor
+      }
+      edges {
+        node {
+          id
+          databaseId
+          title
+          date
+          chineseDate
+          uri
+          link
+          featuredImage{
+            node{
+              sourceUrl
+              caption
+            }
+          }
+          mainAndSubCategory {
+            main
+            sub
+          }
+          categories {
+            edges {
+              node {
+                id
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`,
+        variables: {
+            _category: categoryId,
+        },
+        operationName: 'getListsByCate',
+    };
+}
+
+async function handler(ctx) {
+    const category = ctx.req.param('category') ?? '新聞資訊';
+    const limit = ctx.req.query('limit') ?? 20;
+    const embed = ctx.req.query('embed') === '1';
+    const graphqlUrl = 'https://www.i-cable.com/graphql';
```

Comment:
Since the site built atop WordPress. You can use the [WordPress REST API](https://developer.wordpress.org/rest-api/) instead.

For categories, you can find the category id first using `/wp-json/wp/v2/categories` with the search parameter `slug`. The id for `新聞資訊` is `37`. Then you can list recent posts with the required categories using `/wp-json/wp/v2/posts` with the search parameter `categories`, which will be `37` for `新聞資訊`. If you want plain categories, tags, post header image for each post in the posts API, you can add the `_embed` search parameter to `/wp-json/wp/v2/posts`

To adjust the no. of articles returned to the posts API, you can pass the `per_page` search parameter to `/wp-json/wp/v2/posts`.

This way, you can finish RSS generation in only 2 requests.